src/openms/source/ANALYSIS/OPENSWATH/SpectrumAddition.cpp

```diff
@@ -11,12 +11,48 @@
 #include <OpenMS/KERNEL/MSExperiment.h>
 #include <OpenMS/FILTERING/TRANSFORMERS/LinearResamplerAlign.h>
 #include <OpenMS/ANALYSIS/OPENSWATH/DATAACCESS/DataAccessHelper.h>
+#include <numeric> // std::iota
 
 namespace OpenMS
 {
 
-  OpenSwath::SpectrumPtr SpectrumAddition::addUpSpectra(const std::vector<OpenSwath::SpectrumPtr>& all_spectra,
-      double sampling_rate, bool filter_zeros)
+  void SpectrumAddition::sortSpectrumByMZ(OpenSwath::Spectrum& spec)
+  {
+    // Based off of https://stackoverflow.com/questions/1577475/c-sorting-and-keeping-track-of-indexes/43922758#43922758
+
+    //initialize
+    //std::vector<std::pair<double, Size> > sorted_indices;
+    std::vector<size_t> sorted_indices(spec.getMZArray()->data.size());
+    std::iota(sorted_indices.begin(), sorted_indices.end(), 0);
+
+    // sort indexes based on comparing values in v
+    // using std::stable_sort instead of std::sort
+    // to avoid unnecessary index re-orderings
+    // when v contains elements of equal values
+    auto mzArr = spec.getMZArray()->data.begin();
+    std::stable_sort(sorted_indices.begin(), sorted_indices.end(),
+       [mzArr](size_t i1, size_t i2) {return mzArr[i1] < mzArr[i2];});
+
+    // apply sorting across all arrays
+    for (auto& da : spec.getDataArrays() )
```

Comment:
arent you applying this to the mz array again, effectively first sorting it and then reordering it in the same order you just ordered it ?